go-controller/pkg/ovn/obj_retry.go

```diff
@@ -1050,102 +1090,145 @@ func (oc *Controller) deleteResource(objectsToRetry *retryObjs, obj, cachedObj i
 // re-add them. updateAll forces all objects to be attempted to be retried regardless.
 func (oc *Controller) iterateRetryResources(r *retryObjs, updateAll bool) {
 	r.retryMutex.Lock()
-	defer r.retryMutex.Unlock()
 	now := time.Now()
+	type localRetryEntry struct {
+		key string // the key in the retryObjs map holding retryObjs value
+		// cached newObj holds k8s resource failed during add operation
+		newObj interface{}
+		// cached oldObj holds k8s resource failed during delete operation
+		oldObj interface{}
+		// resource retrieved from the K8s API Server
+		kObj interface{} // Pod retrieved from the K8s API Server
+	}
+	localRetryEntries := make([]*localRetryEntry, 0, len(r.entries))
+	var kObj interface{}
+	var err error
 	for objKey, entry := range r.entries {
-		if entry.ignore {
-			continue
-		}
-		// storing original obj for metrics
-		var initObj interface{}
-		if entry.newObj != nil {
-			initObj = entry.newObj
-		} else if entry.oldObj != nil {
-			initObj = entry.oldObj
-		}
 		// check if we need to create the resource object
 		if entry.newObj != nil {
 			// get the latest version of the resource object from the informer;
 			// if it doesn't exist we are not going to create the new object.
-			obj, err := oc.getResourceFromInformerCache(r.oType, objKey)
+			kObj, err = oc.getResourceFromInformerCache(r.oType, objKey)
 			if err != nil {
 				if kerrors.IsNotFound(err) {
 					klog.Infof("%v %s not found in the informers cache,"+
 						" not going to retry object create", r.oType, objKey)
-					entry.newObj = nil
+					kObj = nil
 				} else {
 					klog.Errorf("Failed to look up %v %s in the informers cache,"+
 						" will retry later: %v", r.oType, objKey, err)
 					continue
 				}
-			} else {
-				entry.newObj = obj
 			}
 		}
+		klog.Infof("Gathered %v resource %s for retry", r.oType, objKey)
+		localRetryEntries = append(localRetryEntries, &localRetryEntry{objKey, entry.newObj, entry.oldObj, kObj})
+	}
+	r.retryMutex.Unlock()
+
+	// Now process the above list of pods that need re-try by holding the lock for each one of them.
+	klog.V(5).Infof("Going to retry resource setup for %d number of resource", len(localRetryEntries))
+
+	wg := &sync.WaitGroup{}
+	for _, lre := range localRetryEntries {
+		wg.Add(1)
+		go func(lre *localRetryEntry) {
```

Comment:
can u move this function in separate function to make the code more readable

Reply:
will do. thanks very much for your comments. will make changes and repush.

Reply:
@msherif1234 repushed. please take a look. thanks